plugin/client_rpc.go

```diff
@@ -720,3 +720,54 @@ func (s *apiRPCServer) LogError(args *Z_LogErrorArgs, returns *Z_LogErrorReturns
 	}
 	return nil
 }
+
+type Z_InstallPluginArgs struct {
+	PluginStreamID uint32
+	B              bool
+}
+
+type Z_InstallPluginReturns struct {
+	A *model.Manifest
+	B *model.AppError
+}
+
+func (g *apiRPCClient) InstallPlugin(file io.Reader, replace bool) (*model.Manifest, *model.AppError) {
+	pluginStreamID := g.muxBroker.NextId()
+
+	go func() {
+		uploadPluginConnection, err := g.muxBroker.Accept(pluginStreamID)
+		if err != nil {
+			log.Print("Plugin failed to upload plugin. MuxBroker could not Accept connection", mlog.Err(err))
+			return
+		}
+		defer uploadPluginConnection.Close()
+		serveIOReader(file, uploadPluginConnection)
+	}()
+
+	_args := &Z_InstallPluginArgs{pluginStreamID, replace}
+	_returns := &Z_InstallPluginReturns{}
+	if err := g.client.Call("Plugin.InstallPlugin", _args, _returns); err != nil {
+		log.Print("RPC call InstallPlugin to plugin failed.", mlog.Err(err))
+	}
+
+	return _returns.A, _returns.B
+}
+
+func (g *apiRPCServer) InstallPlugin(args *Z_InstallPluginArgs, returns *Z_InstallPluginReturns) error {
+	receivePluginConnection, err := g.muxBroker.Dial(args.PluginStreamID)
```

Comment:
Instead of going though the complicated process of setting up the streams you could just do what I did for the PluginHTTP and read into buffers and send a byte array: https://github.com/mattermost/mattermost-server/blob/master/plugin/client_rpc.go#L372

Plugin packages are pretty small so it's likely not an issue.

Reply:
Hey Christopher, thanks for the review!

For this one, I'm thinking the worst scenarios where we might have some public files/docs are getting pushed with the Webapp and having big binaries for the server. 

I'm also thinking about some scenarios while using this API at _com.mattermost.marketplace-addon_ plugin, since it might need to update 10s or 100s of plugins at the same time (tho, pretty unlikely at this stage) with big sizes, in this case we might get out of memory.

While considering all of these, I think, I prefer to have streams over byte slices. And we already have a few streams in the code base, it feels well adopted to me.

/cc @lieut-data